File: misc/minv/autopairs.lua
local M = {}

function M.setup()
  local nparis = require("nvim-autopairs")
  local cond = require("nvim-autopairs.conds")
  local Rule = require("nvim-autopairs.rule")
  nparis.setup({ check_ts = true })
  nparis.add_rules({
    Rule("<", ">")
      :with_pair(cond.none())
      :with_move(function(opts)
        return opts.prev_char:match(">") ~= nil
      end)
      :use_key(">"),
  })
end

return M

File: misc/minv/src/custom/autopairs.ts
export function setup(this: void) {
  const req = require as (this: void, mod: string) => AnyTbl;
  const npairs = req("nvim-autopairs");
  const cond = req("nvim-autopairs.conds");
  const Rule = req("nvim-autopairs.rule") as (
    this: void,
    ...args: any[]
  ) => any;
  npairs.setup({ check_ts: true });
  npairs.add_rules([
    Rule("<", ">")
      .with_pair(cond.none())
      .with_move(function (this: void, opts: any) {
        return opts.prev_char.match(">") != undefined;
      })
      .use_key(">"),
  ]);
}
